### PR TITLE
kpb: add ipc4 support

### PR DESCRIPTION
**Pull request by iganakov**
Add IPC4 support to KPB module keeping 
backward compatibility with IPC3.

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Looks clean - might be worth adding some more inline comments into the PR whilst in progress. KPB in general can do with more comments to help people understand flows.

File: src/audio/kpb.c [hunk 1]
```diff
@@ -157,8 +161,178 @@ static void kpb_lock_init(struct comp_data *kpb)
...
+ */
+static int kpb_get_attribute(struct comp_dev *dev,
+			     uint32_t type,
+			     void *value)
+{
+	struct comp_data *kpb = comp_get_drvdata(dev);
+
+	switch (type) {
+	case COMP_ATTR_BASE_CONFIG:
+		*(struct ipc4_base_module_cfg *)value = kpb->ipc4_cfg.base_cfg;
```
Comment (ranj063) [inline src/audio/kpb.c:255]: this wont align with the module adapter's get_atrribute() callback and will make it difficult to convert

**Comment by ranj063** [comment]
please see https://github.com/thesofproject/sof/pull/6523 where I remove these ops

File: src/audio/kpb.c [hunk 2]
```diff
@@ -1670,6 +1859,11 @@ static const struct comp_driver comp_kpb = {
 		.prepare	= kpb_prepare,
 		.reset		= kpb_reset,
 		.params		= kpb_params,
+#ifdef CONFIG_IPC_MAJOR_4
+		.get_attribute	= kpb_get_attribute,
+		.bind		= kpb_bind,
+		.unbind		= kpb_unbind,
```
[the remark this answers is not in the document]
Comment (iganakov) [reply src/audio/kpb.c:1865]: @ranj063 removing bind/unbind requires additional code refactoring and validation. For now I'd rather left it as is and use legacy component interface to enable KD + KPB flow with IPC4 protocol. I'm going to add module adapter as a next step of upgrading KPB module.

File: src/audio/kpb.c (hunk 1 above)
Comment (lgirdwood) [reply src/audio/kpb.c:255]: @ranj063 what change is needed here ?

**Review by mwasko — APPROVED** [review]

**Pull request merged**